main.go

```diff
@@ -13,9 +13,15 @@ import (
 
 func main() {
 	if err := cmd.Run(context.Background(), os.Args[1:]...); err != nil {
-		if !errors.Is(err, terminal.InterruptErr) {
-			logging.S.Error(err.Error())
+		var userErr cmd.UserFacingError
+		switch {
+		case errors.Is(err, terminal.InterruptErr):
+		case errors.As(err, &userErr):
```

Comment:
```suggestion
		case errors.Is(err, terminal.InterruptErr), errors.As(err, &userErr):
```
I think you can do both of these cases in one line

Reply:
I think that changes the behaviour. Right now we're printing nothing for the interruptErr.

I guess a comment on that line would make it more obvious?

```go
case errors.Is(err, terminal.InterruptErr):
    // print nothing, the user initiated the exit
```

Reply:
Golang doesn't have the same semantics for switch-case as other languages in that fallthrough is not implied. You need to add `fallthrough` to `case errors.Is(err, terminal.InterruptErr):` to mimic the behaviour from other languages. In this case, Bruce's suggestion will change behaviour. Not sure if this was your original intention.

Reply:
original intention was to do nothing when its a terminal error, while do something for the other cases

Reply:
I was also confused at first, Golang has the exact opposite behavior of what I'm familiar with (which is fall through by default)